editor/scene_tree_dock.cpp

```diff
@@ -2195,6 +2195,10 @@ void SceneTreeDock::_do_reparent(Node *p_new_parent, int p_position_in_parent, V
 				undo_redo->add_undo_method(node, "set_position", Object::cast_to<Control>(node)->get_position());
 			}
 		}
+
+		if (Object::cast_to<Container>(new_parent) && Object::cast_to<Control>(node)) {
+			undo_redo->add_undo_method(node, "_edit_set_state", Object::cast_to<Control>(node)->_edit_get_state());
```

Comment:
This a bit conflicts with the Control operation above (`set_position`). When adding node to container, its position is overriden by the set state, so setting it is redundant.